lib/puppet/external/dot.rb

```diff
@@ -196,16 +196,18 @@ def pop
     def to_s(t = '')
       # This code is totally incomprehensible; it needs to be replaced!
 
-      label = @options['shape'] != 'record' && @ports.length == 0 ?
-          @options['label'] ?
-            t + $tab + "label = #{stringify(@options['label'])}\n" :
-            '' :
-          t + $tab + 'label = "' + " \\\n" +
-            t + $tab2 + "#{stringify(@options['label'])}| \\\n" +
-            @ports.collect { |i|
-              t + $tab2 + i.to_s
-            }.join("| \\\n") + " \\\n" +
-            t + $tab + '"' + "\n"
+      label = if @options['shape'] != 'record' && @ports.length == 0
+                @options['label'] ?
```

Comment:
While we're in here updating ternary operators, it might be more readable if you also convert this nested ternary to an `if . . . else . . . end` as well